src/games/supported/Freeway.cpp

```diff
@@ -95,3 +98,38 @@ void FreewaySettings::loadState(Deserializer & ser) {
   m_terminal = ser.getBool();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect FreewaySettings::getAvailableModes() {
+    ModeVect modes(8);
+    for(unsigned i = 0; i < 8; i++) {
+        modes[i] = i;
+    }
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void FreewaySettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment) {
+    if(m < 8) { /*m >= 0 is implicit, since m is an unsigned int*/

```

Comment:
8 shouldn't be hardcoded also. Maybe a static const int at the beginning of the file?

Reply:
Fixed.